core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -703,6 +828,144 @@ object Chunk {
       Doubles(values, 0, values.length)
   }
 
+  /** Concatenates the specified sequence of chunks in to a single chunk, avoiding boxing. */
+  def concat[A](chunks: Seq[Chunk[A]]): Chunk[A] =
+    if (chunks.isEmpty) {
+      Chunk.empty
+    } else if (chunks.forall(c => c.knownElementType[Boolean] || c.forall(_.isInstanceOf[Boolean]))) {
```

Comment:
Huh… I vaguely remember this code. Did I actually write this predicate?

```scala
c.forall(_.isInstanceOf[Boolean])
```

It would be radically more efficient to have a `allTypeOf[Boolean]` check, since that can be answered without actually iterating over the chunk. This is especially important for direct `ByteBuffer` chunks.

Reply:
`c.knownElementType[Boolean]` will return true if `c` is an instance of `Booleans`. Otherwise, we resort to checking instance of each value. How would `allTypeOf` be implemented? If it's specialized in each of the primitive chunk types, then `knownElementType` accomplishes the same thing, right?

Reply:
Ah it's an `||`. Brain similarly not working on this one. You're very correct.